subxt/src/blocks/extrinsic_types.rs

```diff
@@ -605,19 +609,19 @@ impl<T: Config> ExtrinsicEvents<T> {
 
 /// The signed extensions of an extrinsic.
 #[derive(Debug, Clone)]
-pub struct ExtrinsicSignedExtensions<'a> {
+pub struct ExtrinsicSignedExtensions<'a, T: Config> {
     bytes: &'a [u8],
-    metadata: &'a Metadata,
+    metadata: Metadata,
+    _marker: std::marker::PhantomData<T>,
 }
 
-impl<'a> ExtrinsicSignedExtensions<'a> {
+impl<'a, T: Config> ExtrinsicSignedExtensions<'a, T> {
     /// Returns an iterator over each of the signed extension details of the extrinsic.
     /// If the decoding of any signed extension fails, an error item is yielded and the iterator stops.
-    pub fn iter(&self) -> impl Iterator<Item = Result<ExtrinsicSignedExtension<'a>, Error>> {
+    pub fn iter(&'a self) -> impl Iterator<Item = Result<ExtrinsicSignedExtension<'a, T>, Error>> {
```

Comment:
I think you added this `&'a self` lifetime back because metadata is no longer referenced, but you should be able to get rid of it (esp since you `self.metadata.clone()` below still).

If `Metadata` needs to be owned in `ExtrinsicSignedExtensions` (even though other bits are borrowed) then you could:
- `let metadata = self.metadata.clone()` at the top
- then don't reference `self` in the `iter::from_fn` below.

But since everything else is borrowed, I'd borrow the Metadata in `ExtrinsicSignedExtensions` and then all of the data has the `'a` lifetime and that lifetime flows into the iterated stuff too. (Basically, we should either fully commit to lifetimes or avoid them completely here I reckon, and fully committing seems like the good way to go unless we run into some case where they are a pain)